crates/governance/src/event.rs

```diff
@@ -0,0 +1,265 @@
+    }
+
+    /// Create a new proposal event for pgf stewards proposal
+    pub fn pgf_steward_proposal_event(proposal_id: u64, result: bool) -> Self {
+        Self {
+            id: proposal_id,
+            kind: ProposalEventKind::PgfSteward { result },
+        }
+    }
+
+    /// Create a new proposal event for pgf payments proposal
+    pub fn pgf_payments_proposal_event(proposal_id: u64, result: bool) -> Self {
+        Self {
+            id: proposal_id,
+            kind: ProposalEventKind::PgfPayments { result },
+        }
+    }
+}
+
+/// Proposal event kinds.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum ProposalEventKind {
+    /// Governance proposal that has passed.
+    Passed {
+        /// Whether the proposal has WASM code to be executed or not.
+        has_code: bool,
+        /// The execution status of the proposal.
+        execution_status: bool,
+    },
+    /// Governance proposal that has been rejected.
+    Rejected,
+    /// PGF steward governance proposal.
+    PgfSteward {
+        /// The outcome of the proposal.
+        result: bool,
+    },
+    /// PGF payments governance proposal.
+    PgfPayments {
+        /// The outcome of the proposal.
+        result: bool,
+    },
+}
+
+impl From<ProposalEvent> for Event {
+    fn from(proposal_event: ProposalEvent) -> Self {
+        let ProposalEvent {
+            id: proposal_id,
+            kind,
+        } = proposal_event;
+
+        let (event_type, attributes) = match kind {
+            ProposalEventKind::Passed {
+                has_code,
+                execution_status,
+            } => (
+                types::PROPOSAL_PASSED,
+                governance_proposal_attributes(
+                    GovTallyResult::Passed,
+                    proposal_id,
+                    has_code,
+                    execution_status,
+                ),
+            ),
+            ProposalEventKind::Rejected => (
+                types::PROPOSAL_REJECTED,
+                governance_proposal_attributes(
+                    GovTallyResult::Rejected,
+                    proposal_id,
+                    false,
+                    false,
+                ),
+            ),
+            ProposalEventKind::PgfSteward { result } => (
+                types::PROPOSAL_PGF_STEWARD,
+                governance_proposal_attributes(
+                    GovTallyResult::Passed,
+                    proposal_id,
+                    false,
+                    result,
+                ),
+            ),
+            ProposalEventKind::PgfPayments { result } => (
+                types::PROPOSAL_PGF_PAYMENTS,
+                governance_proposal_attributes(
+                    GovTallyResult::Passed,
+                    proposal_id,
+                    false,
+                    result,
+                ),
+            ),
+        };
+
+        let mut event = Self::new(event_type, EventLevel::Block);
+
+        #[allow(deprecated)]
```

Comment:
should we open an issue for replacing these?

Reply:
yeah, ideally